org.envirocar.app/src/org/envirocar/app/views/login/SignupActivity.java

```diff
@@ -222,30 +224,35 @@ private void register(String username, String email, String password) {
 
         registerSubscription = backgroundWorker.schedule(() -> {
             try {
-                User newUser = new UserImpl(username, password);
-                newUser.setMail(email);
-                daoProvider.getUserDAO().createUser(newUser);
-
-                // Successfully created the getUserStatistic
-                mainThreadWorker.schedule(() -> {
-                    // Dismiss the progress dialog.
+                if(new ContextInternetAccessProvider(getApplicationContext()).isConnected()) {
```

Comment:
I would suggest to place the [these lines](https://github.com/enviroCar/enviroCar-app/blob/0c4fa0d8632ba9aa6898282748a6efe2bf536d02/org.envirocar.app/src/org/envirocar/app/views/login/SignupActivity.java#L219-L223) inside the if-statement, so that the dialog won't flacker for a few milliseconds